# Importamos librerias  	


In [2]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)

# Importamos BB.DD

In [3]:
df_orig = pd.read_csv("datos/netflix_originals.csv", index_col=0)
df_titles= pd.read_csv("datos/netflix_titles.csv",index_col=0)

# 2. Analisis de columnas y consistencias

## Netflix_Titles - df_titles

### Visualizacion general del df

In [4]:
df_titles

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA        NaN   
3     September 24, 2021          2021  TV-MA        NaN   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R        NaN   
8805    January 11, 2020          2006     PG        NaN   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

### Visualizacion info especifica del df

In [5]:
df_titles.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      3994 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 894.5+ KB


- Con la funcion info podemos observar que exites 8807 peliculas, series y documentales. 
- La mayoria de las columnas son de tipo objeto, menos el año de lanzamiento. Lo curioso es que la duracion esta puesta como objeto(string), al igual que la fecha de lanzamiento. 
- Tenemos muchos datos nulos en lo que es directores y ni hablar a lo que se refiere en la duracion.

### Hacemos un analisis descriptivo de los titulos

In [25]:
df_titles.describe(include=['object', 'category'])

show_id   type   title       director                cast  \
count     8807   8807    8807           6173                7982   
unique    8807      2    8807           4528                7692   
top      s8807  Movie  Zubaan  Rajiv Chilaka  David Attenborough   
freq         1   6131       1             19                  19   

              country       date_added rating   duration  \
count            7976             8797   8803       3994   
unique            748             1767     17        185   
top     United States  January 1, 2020  TV-MA  2 Seasons   
freq             2818              109   3207        425   

                           listed_in  \
count                           8807   
unique                           514   
top     Dramas, International Movies   
freq                             362   

                                              description  
count                                                8807  
unique                                               8775  
top     Paranormal activity at a lush, abandoned prope...  
freq                                                    4

- podemos ver que existen unicamente 2 tipos de contenidos -> peliculas y series
- titulos todos diferentes por lo que no hay valores duplicados
- directores hay 4528 que tienen su contenido en plataforma
- Respecto a los paises el que mayor contenido produce es estados unidos con 2818. 
- el genero con mayor contenido es lo que son los dramas internacionales.

In [39]:
df_titles.describe().T

count         mean       std     min     25%     50%     75%  \
release_year  8807.0  2014.180198  8.819312  1925.0  2013.0  2017.0  2019.0   

                 max  
release_year  2021.0

Respecto a variable numerica a analizar solo contamos con el año de lanzamiento
- La media dio en 2014 pero la mediana es 2017
- El 75% del contenido se lanzo antes de 2019.

### Checkeamos nulos de df_titles

- ### Creamos funcion buscar_nulos() para poder pasar dataframe que quiseramos y asi encontrar aquellos valores faltantes en cada columna

In [7]:
def buscar_nulos(df):
    #Creamos Df vacio
    df_titulos_nulos=pd.DataFrame()

    df_titulos_nulos["Numero_Nulos"] = df.isnull().sum() #Creamos columna de nulos -> si isnull()=True= 1 => si sumamos todos, nos da el numero de nulos
    df_titulos_nulos["porcentaje_nulos"] = round((df.isnull().sum()/df.shape[0])*100,2) #Para saber el porcenaje de nulos, lo dividimos sobre la cantidad de filas que tiene nuesto df
    return df_titulos_nulos.T



In [8]:
buscar_nulos(df_titles) 	

show_id  type  title  director    cast  country  date_added  \
Numero_Nulos          0.0   0.0    0.0   2634.00  825.00   831.00       10.00   
porcentaje_nulos      0.0   0.0    0.0     29.91    9.37     9.44        0.11   

                  release_year  rating  duration  listed_in  description  
Numero_Nulos               0.0    4.00   4813.00        0.0          0.0  
porcentaje_nulos           0.0    0.05     54.65        0.0          0.0

- Casi el 30% de la categoria director no tenemos conocimiento de quien hizo ese contenido
- Otro dato grave que desconocemos es el la duracion del contenido. Mas del 54% desconocemos cuanto dura.
- Con valores similares, el reparto y de que pais proviene el contenido, tienen un poco mas del 9% de nulos. 

### Checkeamos duplicados de df_titles

#### Creamos funcion para reconocer duplicados

In [9]:
def duplicados(df):
    return df.duplicated().sum()

### Buscando en todo el data frame de titulos.

In [10]:
duplicados(df_titles)

np.int64(0)

- No contamos con valores duplicados en la base de datos de titulos de netflix.

### Buscando especificamente si hay titulos repetidos

In [11]:
duplicados(df_titles["title"])

np.int64(0)

- Tampoco contamos con duplicados en lo que es especificamente los titulos. Es decir, no tenemos ningun titulo que se repite de los 8807.

## Creamos funcion para ver los valores unicos por columna

In [12]:
def valores_unicos_columnas(df, columna=None):
    #creamos df vacio
    df_val_unico=pd.DataFrame()
    # llenamos el df con los tiposs y valores
    df_val_unico["tipo"] = df[columna].unique()
    df_val_unico["cantidad"] = df[columna].value_counts().values

    return df_val_unico

## Vemos que tipo de contenido es el mayor subido

In [27]:
valores_unicos_columnas(df_titles, columna="type")

tipo  cantidad
0    Movie      6131
1  TV Show      2676

- la mayoria del contenido de la bbdd de titulos son peliculas

# Conclusiones EDA -> df_titles
- Existen columnas de datos que no tienen el tipo correcto de dato. Por ejemplo: Duration(hay algunos que dicen 2 seasons) y fecha de agregacion (esta en formato largo) a la plataforma
- Tomar accion con los nulos en df_titles
    -   Por del 54% no tenemos conocimiento de cuanto dura el contenido. Altamente grave ya que no podemos saber que prefiere la gente cuando se trata de duracion.
    -   Casi el 30% del contenido no tiene directores. 
    -   Casi el 10% del contenido no tiene el reparto ni de que pais es el mismo.
    -   Podemos omitir el contenido date_added y rating rellenandolo con la mediana de cada uno, o la moda. 
- No contamos con datos duplicados en la bbdd ya que todos los titulos son diferentes
- Existen unicamente 2 tipos de contenidos -> peliculas y series
- Contamos con informacion de que hay 4528 directores en la plataforma
- Respecto a los paises, el que mayor contenido produce es EEUU con 2818. 
- El genero con mayor contenido es lo que son los dramas internacionales.


## Netflix_originals - df_orig


### Visualizacion general del df

In [14]:
df_orig

Title                  Genre  \
0                                Enter the Anime            Documentary   
1                                    Dark Forces               Thriller   
2                                        The App  Science fiction/Drama   
3                                 The Open House        Horror thriller   
4                                    Kaali Khuhi                Mystery   
..                                           ...                    ...   
576                       Cuba and the Cameraman            Documentary   
577                       Dancing with the Birds            Documentary   
580  Winter on Fire: Ukraine's Fight for Freedom            Documentary   
581                      Springsteen on Broadway           One-man show   
583     David Attenborough: A Life on Our Planet            Documentary   

              Premiere  Runtime  IMDB Score                  Language  
0       August 5, 2019       58         2.5          English/Japanese  
1      August 21, 2020       81         2.6                   Spanish  
2    December 26, 2019       79         2.6                   Italian  
3     January 19, 2018       94         3.2                   English  
4     October 30, 2020       90         3.4                     Hindi  
..                 ...      ...         ...                       ...  
576  November 24, 2017      114         8.3                   English  
577   October 23, 2019       51         8.3                   English  
580    October 9, 2015       91         8.4  English/Ukranian/Russian  
581  December 16, 2018      153         8.5                   English  
583    October 4, 2020       83         9.0                   English  

[513 rows x 6 columns]

### Visualizacion info especifica del df

In [15]:
df_orig.info()

<class 'pandas.core.frame.DataFrame'>
Index: 513 entries, 0 to 583
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       513 non-null    object 
 1   Genre       513 non-null    object 
 2   Premiere    513 non-null    object 
 3   Runtime     513 non-null    int64  
 4   IMDB Score  513 non-null    float64
 5   Language    513 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 28.1+ KB


Respecto a los titulos originales de netflix podemos detacar que:
- No se encontraron nulos
- Son un total de 513 peliculas, series y documentales
- Tenemos la duracion de todos los titulos que en el df posterior no teniamos de muchos.

### Hacemos un analisis descriptivo de los titulos

In [31]:
df_orig.describe(include=['object', 'category']).T

count unique                                       top freq
Title      513    513  David Attenborough: A Life on Our Planet    1
Genre      513    106                               Documentary  132
Premiere   513    350                          October 18, 2019    5
Language   513     37                                   English  352

- Destacamos lo relevante que son los documentales en los contenidos exclusivos de netflix 
- Del total de 513 elementos, 352 son en ingles. 
- El dia con mayores lanzamientos en la plataforma fue el 18/10/2019

In [37]:
df_orig.describe().T

count       mean        std  min   25%   50%    75%    max
Runtime     513.0  94.674464  26.648121  4.0  87.0  97.0  108.0  209.0
IMDB Score  513.0   6.210916   0.968850  2.5   5.6   6.3    6.9    9.0

Respecto a las variables numericas:
- La nota de IMDB: 
    - La media del contenido de netflix es de 6,21 estrellas, esto muy similar a la mediana de 6,3. 
    - Casi el 25% del contenido de netflix tiene una nota superior a 7 estrellas, por lo que el contenido que produce netflix es muy mediocre.
- La duracion del contenido:
    - Media y mediana muy similares en tiempo, entre 94 y 97 minutos respectivamente
    - El 75% de las peliculas no llegan a las 2horas. Es decir en su mayoria son cortas.


### Chequeamos nulos con funcion declarada previamente

In [16]:
buscar_nulos(df_orig)

Title  Genre  Premiere  Runtime  IMDB Score  Language
Numero_Nulos        0.0    0.0       0.0      0.0         0.0       0.0
porcentaje_nulos    0.0    0.0       0.0      0.0         0.0       0.0

Como vimos en el cuadro anterior, no existian valores nulos.

### Chequeamos duplicados con funcion declarada previamente

In [17]:
duplicados(df_orig)

np.int64(0)

No existen valores duplicados.

In [35]:
valores_unicos_columnas(df_orig, columna="Genre")


tipo  cantidad
0                             Documentary       132
1                                Thriller        73
2                   Science fiction/Drama        42
3                         Horror thriller        35
4                                 Mystery        33
..                                    ...       ...
101                      Action-adventure         1
102                          Concert Film         1
103           Animation / Science Fiction         1
104                 Anthology/Dark comedy         1
105  Animation/Christmas/Comedy/Adventure         1

[106 rows x 2 columns]

# Conclusiones EDA -> df_orig

- 513 elementos originales de netflix
- Los documentales de netflix son los mas populares, seguidos por los thrillersm 
- Del total de 513 elementos, 352 son en idioma de ingles. 
- El dia con mayores lanzamientos en la plataforma fue el 18/10/2019
- No se encontraron valores duplicados ni nulos.
Respecto a las variables numericas:
- La media del contenido de netflix es de 6,21 estrellas, esto muy similar a la mediana de 6,3. 
- Casi el 25% del contenido de netflix tiene una nota superior a 7 estrellas, por lo que el contenido que produce netflix es muy mediocre.
- La duracion Media y mediana muy similares en tiempo, entre 94 y 97 minutos respectivamente
- El 75% de las peliculas no llegan a las 2horas. Es decir en su mayoria son cortas.